IPython/terminal/prompts.py

```diff
@@ -6,6 +6,11 @@
 
 from IPython.core.displayhook import DisplayHook
 
+try:
+    from prompt_toolkit.layout.utils import token_list_width
+except ImportError:
+    token_list_width = lambda in_tokens: sum(len(s) for (t, s) in in_tokens)
```

Comment:
The function is there in PTK 1.0.0, so there should be no need for a fallback.

https://github.com/jonathanslenders/python-prompt-toolkit/blob/1.0.0/prompt_toolkit/layout/utils.py#L25